nucypher/blockchain/eth/sol/source/contracts/MinersEscrow.sol

```diff
@@ -638,7 +713,7 @@ contract MinersEscrow is Issuer {
                 continue;
             }
             if (addMoreTokens) {
-                sumOfLockedTokens = sumOfLockedTokens.add(getLockedTokens(currentMiner, _periods));
+                sumOfLockedTokens = sumOfLockedTokens.add(getLockedTokens(currentMiner, int16(_periods)));
```

Comment:
We need negative numbers? Why `int16`?

Reply:
Is it possible to set `_periods` larger than 16 bits?

Reply:
16 bits signed int will give 89 years of operation. I think, we'd need to deal with Unix timestamp maximum date before that

Reply:
Hardfork 2107 :joy:

Reply:
Local mainnet on Mars is operational

Reply:
`uint16` -> `int16` cast was removed. All overflow while addition and subtraction are controlled by `SafeMath` library